quinn-proto/src/connection/mod.rs

```diff
@@ -2575,6 +2583,18 @@ where
             }
         }
 
+        // Issue stream ID credit due to ACKs of outgoing finish/resets and incoming finish/resets
```

Comment:
Are early returns an issue with having it here? I guess likely not, because those are all connection errors and in that case there won't be new streams anyway?

Reply:
Correct; any early return here is a transport error, and transport errors always immediately terminate the connection, rendering flow control moot.